### PR TITLE
add readthedocs.yml

### LINKED ISSUE #182: enable PR builds for readthedocs
@hynek @psi29a can you enable PR builds for https://docs.readthedocs.io/en/stable/guides/autobuild-docs-for-pull-requests.html https://readthedocs.org/projects/ldaptor/ please?

can you also export the current settings as a .readthedocs.yml  file?

### PR DESCRIPTION
**Pull request by graingert**
Fixes #182 

# Contributor Checklist:

* [ ] I have updated the release notes at `docs/source/NEWS.rst`
* [x] I have updated the automated tests.
* [ ] All tests pass on your local dev environment. See `CONTRIBUTING.rst`.

### CODE REVIEW

**Comment by graingert** [comment]
@psi29a can you add docs/readthedocs.org:ldaptor to the required checks?

File: setup.cfg [hunk 1]
```diff
@@ -48,6 +48,18 @@ scripts =
...
+docs =
+  alabaster~=0.7.12
+  commonmark~=0.9.1
+  docutils~=0.16.0
+  mock~=4.0
+  pillow~=7.2
+  readthedocs-sphinx-ext~=2.1
+  recommonmark~=0.6.0
+  sphinx~=3.2
+  sphinx-rtd-theme~=0.5.0
```
Comment (graingert) [inline setup.cfg:61]: based on the packages readthedocs installs, but upgraded:

``` /home/docs/checkouts/readthedocs.org/user_builds/ldaptor/envs/185/bin/python -m pip install --upgrade --no-cache-dir --use-feature 2020-resolver setuptools==41.0.1 docutils==0.14 mock==1.0.1 pillow==5.4.1 alabaster>=0.7,<0.8,!=0.7.5 commonmark==0.8.1 recommonmark==0.5.0 sphinx<2 sphinx-rtd-theme<0.5 readthedocs-sphinx-ext<1.1 ```

**Comment by adiroiban** [comment]
read the docs should now be a required check.

I think that the default readme should be updated to link to read the docs page.

Also, it would be nice to have a badge for latest pypi release inside the README.

Thanks!

**Comment by graingert** [comment]
@adiroiban please can you approve this PR it's ready and blocking all other PRs

**Comment by graingert** [comment]
@adiroiban can you add  "CI / all-successful" as a required check

**Review by adiroiban — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**